lib/uniffi-zcash-cli/src/cli.rs

```diff
@@ -206,7 +248,23 @@ pub fn validator_regex(regex: &'static str, err_msg: &'static str) -> ValueParse
         let reg = regex::Regex::new(regex).unwrap();
         match reg.is_match(input) {
             true => Ok(input.to_owned()),
-            false => Err(anyhow!("Value \"{}\" is not matching format: {}", input, err_msg)),
+            false => Err(anyhow!(
+                "Value \"{}\" is not matching format: {}",
+                input,
+                err_msg
+            )),
+        }
+    })
+}
+
+/// Validates if a path exists
+pub fn validator_existing_path() -> ValueParser {
+    ValueParser::from(move |input: &str| -> anyhow::Result<String> {
+        let exists = Path::is_dir(Path::new(input));
+        if exists {
+            Ok(input.to_string())
+        } else {
+            Err(anyhow!("path \"{}\" not found", input))
         }
```

Comment:
```suggestion
        ensure!(Path::new(input).is_dir(), "path \"{input}\" not found");
        Ok(input.to_string())
```

Reply:
you can write that simpler with `anyhow::ensure!`. Also whenever you'll need writing something like
```
return Err(anyhow!("fooo"));
```
then there is also a convenience macro for that
```
bail!("foo")
```